From 2a87f77dcd5a6e4b8bfae3225737548ba69fafcc Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 19 Oct 2021 19:55:59 +0200
Subject: [PATCH 1/2] Fix import and fail install

1) Fix the import until upstream releases a new version.
2) Do not install dependencies if not available.
---
 cmake/python.cmake | 20 ++++++++------------
 1 file changed, 8 insertions(+), 12 deletions(-)

diff --git a/cmake/python.cmake b/cmake/python.cmake
index a19d2d2ab3..5c77a60669 100644
--- a/cmake/python.cmake
+++ b/cmake/python.cmake
@@ -27,10 +27,10 @@ find_package(Python3 REQUIRED COMPONENTS Interpreter Development.Module)
 list(APPEND CMAKE_SWIG_FLAGS "-py3" "-DPY3")

 # Find if python module MODULE_NAME is available,
-# if not install it to the Python 3 user install directory.
-function(search_python_module MODULE_NAME)
+# if not install it (PACKAGE_NAME) to the Python3 user install directory.
+function(search_python_module MODULE_NAME PACKAGE_NAME)
   execute_process(
-    COMMAND ${Python3_EXECUTABLE} -c "import ${MODULE_NAME}; print(${MODULE_NAME}.__version__)"
+    COMMAND ${Python3_EXECUTABLE} -c "import ${MODULE_NAME}; print(${MODULE_NAME}.__package__)"
     RESULT_VARIABLE _RESULT
     OUTPUT_VARIABLE MODULE_VERSION
     ERROR_QUIET
@@ -39,11 +39,7 @@ function(search_python_module MODULE_NAME)
   if(${_RESULT} STREQUAL "0")
     message(STATUS "Found python module: ${MODULE_NAME} (found version \"${MODULE_VERSION}\")")
   else()
-    message(WARNING "Can't find python module \"${MODULE_NAME}\", install it using pip...")
-    execute_process(
-      COMMAND ${Python3_EXECUTABLE} -m pip install --user ${MODULE_NAME}
-      OUTPUT_STRIP_TRAILING_WHITESPACE
-      )
+    message(FATAL_ERROR "Can't find python module \"${MODULE_NAME}\", please install it using your system package manager.")
   endif()
 endfunction()

@@ -64,7 +60,7 @@ function(search_python_internal_module MODULE_NAME)
 endfunction()

 # Generate Protobuf py sources with mypy support
-search_python_module(mypy-protobuf)
+search_python_module(mypy_protobuf mypy-protobuf)
 set(PROTO_PYS)
 file(GLOB_RECURSE proto_py_files RELATIVE ${PROJECT_SOURCE_DIR}
   "ortools/constraint_solver/*.proto"
@@ -157,8 +153,8 @@ configure_file(
   COPYONLY)

 # Look for python module wheel
-search_python_module(setuptools)
-search_python_module(wheel)
+search_python_module(setuptools setuptools)
+search_python_module(wheel wheel)

 # Main Target
 add_custom_target(python_package ALL
@@ -199,7 +195,7 @@ install(SCRIPT ${PROJECT_BINARY_DIR}/python/python-install.cmake)
 # Test
 if(BUILD_TESTING)
   # Look for python module venv
-  search_python_module(absl-py)
+  search_python_module(absl absl-py)
   search_python_internal_module(venv)
   # Testing using a vitual environment
   set(VENV_EXECUTABLE ${Python3_EXECUTABLE} -m venv)
--
2.33.0
